### PR TITLE
Propose new MaxPool-8 operation

### PR DESCRIPTION
**Pull request by postrational**
### Details:

Specification of MaxPool operation changes to match ONNX operation specification.

### CODE REVIEW

File: docs/ops/pooling/MaxPool_1.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_1.md:21]: For the MaxPool-8 we also want to add a new attribute defining the padding value. This change is needed to resolve issue that TF/ONNX and Caffe have different default value for the padded elements (TF/ONNX - 0, while Caffe - minus infimum). Some details in the ticket 35367

**Comment by postrational** [comment]
Suggestion: also add attributes to AvgPool

**Comment by mvafin** [comment]
Let's align these changes with #5773.

File: docs/ops/pooling/MaxPool_1.md
Comment (postrational) [inline docs/ops/pooling/MaxPool_1.md:34]: Suggestion: change default to `-inf`

File: docs/ops/pooling/MaxPool_1.md
Comment (postrational) [inline docs/ops/pooling/MaxPool_1.md:105]: Comment: second output may only be computed correctly if `pads_value` is set to the default `-inf`

File: docs/ops/pooling/MaxPool_1.md
Comment (postrational) [reply docs/ops/pooling/MaxPool_1.md:21]: Added `pads_value`

File: docs/ops/pooling/MaxPool_8.md
Comment (tomdol) [inline docs/ops/pooling/MaxPool_8.md:23]:
```suggestion
    * **Description**: *dilations* specify the index of the next pixel to select when pooling. If not present, the dilation defaults to 1, meaning the adjacent pixel is chosen. A value of 2 indicates that one pixel is skipped and every other pixel is considered. Dilations specify one value for each spatial axis of the kernel: `(z, y, x)` for 3D poolings and `(y, x)`  for 2D poolings.
```

**Review by tomdol — CHANGES_REQUESTED** [review]
I think the examples should also contain the values of the second output of the new operator too.

**Comment by tomdol** [comment]
To match the PDPD behavior, please extend the spec with an additional attribute `axis`.  It should point to the dimension of the data input shape which should be used as a starting point to calculate the upper bound of allowed indices values.

```
Input shape: {N, C, H, W, D}
axis = 0   =>  indices range: [0; N*C*H*W*D)
axis = 2   =>  indices range: [0; H*W*D)
```

The axis has to point to one of the existing dimensions, consider if negative values should be allowed.

File: docs/ops/pooling/MaxPool_1.md
Comment (igorsafo) [reply docs/ops/pooling/MaxPool_1.md]: @lazarevevgeny @postrational Could you please modify `pads_value` to be a string instead of float value to limit space of possible values? As I understand there is no reason for supporting anything except for `0` and `-inf`. Also float padding doesn't make any sense for pooling on integer data, but it will complicate implementation a lot to support such cases.

```
* *pads_value*

    * **Description**: *pads_value* will be used to fill pixels added during padding. If not specified, the new pixels will be filled in with minimum value for the type.
    * **Range of values**:
      * *MIN*
      * *ZERO*
    * **Type**: string
    * **Default value**: *MIN*
    * **Required**: *no*
```

Comment (igorsafo) [reply docs/ops/pooling/MaxPool_1.md]: @lazarevevgeny Also could you please clarify the issue? As I understand current IE and oneDNN implementation is aligned with TF (pad: -inf):
```python
>>> import tensorflow as tf
>>> import numpy as np
>>> a = np.array([-0.8, -0.3, 0.4,
...               -0.1, -0.2, 0.3,
...               -1.0, -2.0, -3.0])
>>> a = np.reshape(a, (1, 3, 3, 1))
>>> x = tf.constant(a)
>>> y = tf.nn.max_pool2d(x, ksize=[3, 3], strides=[2, 2], padding='SAME')
>>> y
<tf.Tensor: shape=(1, 2, 2, 1), dtype=float64, numpy=
array([[[[-0.1],
         [ 0.4]],

        [[-0.1],
         [ 0.3]]]])>
```

**Review by tomdol — APPROVED** [review]

**Comment by tomdol** [comment]
@lazarevevgeny I've modified the spec a lot to include the dilations attribute in the output shape calculations. I've added the second output values to the examples and addressed the `axis` as well. Can you please perform the review again?

File: docs/ops/pooling/MaxPool_8.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_8.md:18]: We do not add "Default value" if the attribute is "Required" anymore.

File: docs/ops/pooling/MaxPool_8.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_8.md:95]: Change `[-R, R-1]` to `[-R, R - 1]`

File: docs/ops/pooling/MaxPool_8.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_8.md:109]: We usually write `1D` instead of `1-D`, aren't we?

File: docs/ops/pooling/MaxPool_8.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_8.md:140]: Add spaces around `/`

File: docs/ops/pooling/MaxPool_8.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_8.md:213]: You have fixed a bug in the specification. Please, update the MaxPool_1.md as well.

File: docs/ops/pooling/MaxPool_8.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_8.md:26]: wrap with "`" characters

**Review by lazarevevgeny — APPROVED** [review]
Looks good. Just minor comments. Approving in advance.

File: docs/ops/pooling/MaxPool_8.md
Comment (avladimi) [inline docs/ops/pooling/MaxPool_8.md:144]:
```suggestion
Example 1 shows how *MaxPool* operates with 4D input using 2D kernel and `auto_pad = explicit`.
```

File: docs/ops/pooling/MaxPool_8.md
Comment (avladimi) [inline docs/ops/pooling/MaxPool_8.md:166]:
```suggestion
Example 2 shows how *MaxPool* operates with 3D input using 1D kernel and `auto_pad = valid`.
```

File: docs/ops/pooling/MaxPool_8.md
Comment (avladimi) [inline docs/ops/pooling/MaxPool_8.md:178]:
```suggestion
Example 3 shows how *MaxPool* operates with 4D input using 2D kernel and `auto_pad = same_lower`.
```

File: docs/ops/pooling/MaxPool_8.md
Comment (avladimi) [inline docs/ops/pooling/MaxPool_8.md:196]:
```suggestion
Example 4 shows how *MaxPool* operates with 4D input using 2D kernel and `auto_pad = same_upper`.
```

File: docs/ops/pooling/MaxPool_8.md
Comment (avladimi) [inline docs/ops/pooling/MaxPool_8.md:223]:
```suggestion
Example 5 shows how *MaxPool* operates with 4D input using 2D kernel, `auto_pad = valid` and `rounding_type = ceil`.
```

File: docs/ops/pooling/MaxPool_8.md
Comment (avladimi) [inline docs/ops/pooling/MaxPool_8.md:239]:
```suggestion
Example 6 shows how *MaxPool* operates on 4D input using dilated 2D kernel, `auto_pad = explicit` and `rounding_type = floor`.
```

File: docs/ops/pooling/MaxPool_8.md
Comment (avladimi) [inline docs/ops/pooling/MaxPool_8.md:260]:
```suggestion
Example 7 shows how *MaxPool* operates on 4D input using 2D kernel, with non-default `axis` value.
```

**Comment by tomdol** [comment]
Adding the "do not merge" label temporarily due to an offline discussion about the `pads_value` attribute.

File: docs/ops/pooling/MaxPool_8.md
Comment (tomdol) [inline docs/ops/pooling/MaxPool_8.md:28]: Suggestion to add a requirement that the value needs to fit in the data type of the operator's input. Otherwise the op might produce undefined behavior.

File: docs/ops/pooling/MaxPool_8.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_8.md:9]: let's remove references to third-party sites.

File: docs/ops/pooling/MaxPool_8.md
Comment (lazarevevgeny) [inline docs/ops/pooling/MaxPool_8.md:38]: May be add somewhere in the documentation that the padded values are equal to -inf?

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**
